extensions/prototype-content-with-issues-table/app.R

```diff
@@ -0,0 +1,152 @@
+library(shiny)
+library(bslib)
+library(DT)
+library(connectapi)
+library(dplyr)
+library(purrr)
+
+# cache data to disk with a refresh every 8h
+shinyOptions(
+  cache = cachem::cache_disk("./app_cache/cache/", max_age = 60 * 60 * 8)
+)
+
+source("get_usage.R")
+
+# Hacky function to get a list of Content class objects without making a request
+# for each item. These objects differ from the ones created by `content_item()`
+# because they also include the full owner info as returned by `get_content()`.
+as_content_list <- function(content_df, client) {
+  cdf_split <- split(content_df, 1:nrow(content_df))
+  map(cdf_split, function(x) {
+    x <- x[, !(names(x) %in% c("tags"))]
+    x <- as.list(x)
+    Content$new(client, x)
+  })
+}
+
+# checks to see if a content item has failed jobs, grabs usage data if it does,
+# then compiles content, job, and usage data together, returning it.
+get_failed_job_data <- function(item, usage) {
+  failed_jobs <- tryCatch(
+    {
+      get_jobs(item) |> 
+        # filter successful jobs
+        filter(exit_code != 0) |> 
+        # map content job types to something more readable 
+        mutate(tag = case_when(
+               tag %in% c("build_report", "build_site", "build_jupyter") ~ "Building",
+               tag %in% c("packrat_restore", "python_restore") ~ "Restoring environment",
+               tag == "configure_report" ~ "Configuring report",
+               tag %in% c("run_app", 
+                          "run_api", 
+                          "run_tensorflow", 
+                          "run_python_api",
+                          "run_dash_app",
+                          "run_gradio_app",
+                          "run_streamlit",
+                          "run_bokeh_app",
+                          "run_fastapi_app",
+                          "run_voila_app",
+                          "run_pyshiny_app") ~ "Running",
+               tag == "render_shiny" ~ "Rendering",
+               tag == "ctrl_extraction" ~ "Extracting parameters",
+               TRUE ~ tag)) |>
```

Comment:
What do you think about moving this munging into the display code rather than the querying code?